### PR TITLE
Bugfixes

### PR DESCRIPTION
**Pull request by NomAnor**
I was debugging a segfault and found a reference counting bug in the handler functions.

I also corrected a couple minor things.

### CODE REVIEW

File: pyca/pyca.cc [hunk 1]
```diff
@@ -731,6 +731,9 @@ extern "C" {
     // Initialize python module
     DECLARE_INIT(pyca)
     {
+        // libca creates non-python threads so we must ensure that python threading is initialized
+        PyEval_InitThreads();
```
Comment (ZLLentz) [inline pyca/pyca.cc]: This is already called below, before the first epics-related call. Is there a special need to call it earlier?
https://github.com/slaclab/pyca/blob/5e18525fcd2b39c68c5d8dd4b543d4d8e12dd5f2/pyca/pyca.cc#L791-L793

**Review by ZLLentz — COMMENTED** [review]
Hi Andre, thank you for contributing. Good catches here. See my comment about `PyEval_InitThreads`.

**Comment by NomAnor** [comment]
I totally missed the `PyEval_InitThreads` call. I removed the commit and updated the pull-requests.

**Comment by ZLLentz** [comment]
It's all good, you've noticed a lot of things that we've missed for a while. I'm going to merge both of these and tag a bugfix release tomorrow if nobody else objects.

**Pull request merged**